### PR TITLE
CV-157 Add links to manage reservations

### PR DESCRIPTION
**Pull request by iangledhill**
There is no permission toggle around the new links as agreed with team but these are subject to a feature toggle.

### CODE REVIEW

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Authorization/AuthorizationContextProvider.cs [hunk 1]
```diff
@@ -54,7 +55,12 @@ public IAuthorizationContext GetAuthorizationContext()
...
         private long? GetProviderId(RouteValueDictionary routeValueDictionary)
         {
-            if (long.TryParse((string) routeValueDictionary[RouteDataKeys.ProviderId], out var providerId))
+            long providerId;
+
+            if (long.TryParse(_httpContext.User.Identity.GetClaim("http://schemas.portal.com/ukprn"), out providerId))
+                return providerId;
+
+            if (long.TryParse((string) routeValueDictionary[RouteDataKeys.ProviderId], out providerId))
                 return providerId;
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Authorization/AuthorizationContextProvider.cs]: It is always a worry that there are no tests for much of the web layer (but this isn't an issue created here)

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Helpers/HtmlExtensions.cs [hunk 2]
```diff
@@ -3,18 +3,15 @@
 using System.Web.Mvc;
 using SFA.DAS.ProviderApprenticeshipsService.Domain.Models.FeatureToggles;
 using SFA.DAS.ProviderApprenticeshipsService.Web.Extensions;
+using SFA.DAS.Authorization.ProviderPermissions;
```
[the remark this answers is not in the document]
Comment (iangledhill) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Helpers/HtmlExtensions.cs]: Removed

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Helpers/HtmlExtensions.cs [hunk 3]
```diff
@@ -3,18 +3,15 @@
...
 using System;
+using SFA.DAS.Authorization;
```
[the remark this answers is not in the document]
Comment (iangledhill) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Helpers/HtmlExtensions.cs]: Removed

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Authorization/AuthorizationContextProvider.cs (hunk 1 above)
Comment (iangledhill) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Authorization/AuthorizationContextProvider.cs]: There _are_ tests for this class.

**Pull request merged**